<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MetronomeWindow</class>
 <widget class="QMainWindow" name="MetronomeWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>330</width>
    <height>172</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NSMetronome</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QLabel" name="bpmLabel">
        <property name="text">
         <string>Tempo:</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QSpinBox" name="bpmSpinBox">
        <property name="accelerated">
         <bool>true</bool>
        </property>
        <property name="suffix">
         <string> bpm</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnTapTempo">
        <property name="text">
         <string>Tap</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <widget class="QSubdivisionButton" name="btnQuarterth">
          <property name="text">
           <string>1/4</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSubdivisionButton" name="btnEighth">
          <property name="text">
           <string>1/8</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QSubdivisionButton" name="btnSixteenth">
          <property name="text">
           <string>1/16</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSubdivisionButton" name="btnEighthTriplets">
          <property name="text">
           <string>1/8T</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QSubdivisionButton" name="btnSixteenthTriplets">
          <property name="text">
           <string>1/16T</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QSubdivisionButton" name="btnQuarterthTriplets">
          <property name="text">
           <string>1/4T</string>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
          <attribute name="buttonGroup">
           <string notr="true">bgDivisions</string>
          </attribute>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QPushButton" name="btnHalfTempo">
          <property name="text">
           <string>Â½x</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnDoubleTempo">
          <property name="text">
           <string>2x</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QPushButton" name="btnStartStop">
      <property name="text">
       <string>Start</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>QSubdivisionButton</class>
   <extends>QPushButton</extends>
   <header location="global">qsubdivisionbutton.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>btnStartStop</tabstop>
  <tabstop>bpmSpinBox</tabstop>
  <tabstop>btnTapTempo</tabstop>
  <tabstop>btnQuarterth</tabstop>
  <tabstop>btnQuarterthTriplets</tabstop>
  <tabstop>btnEighth</tabstop>
  <tabstop>btnEighthTriplets</tabstop>
  <tabstop>btnSixteenth</tabstop>
  <tabstop>btnSixteenthTriplets</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>btnStartStop</sender>
   <signal>toggled(bool)</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>startStopMetronome(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>98</x>
     <y>155</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bpmSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>changeTempo(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>154</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>86</x>
     <y>41</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>bgDivisions</sender>
   <signal>buttonToggled(QAbstractButton*,bool)</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>chooseSubdivision(QAbstractButton*,bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>182</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnTapTempo</sender>
   <signal>clicked()</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>tapTempo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>317</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>147</x>
     <y>171</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>MetronomeWindow</sender>
   <signal>tempoChanged(int)</signal>
   <receiver>bpmSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>329</x>
     <y>117</y>
    </hint>
    <hint type="destinationlabel">
     <x>159</x>
     <y>27</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnHalfTempo</sender>
   <signal>clicked()</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>halfTempo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>290</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>164</x>
     <y>911</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>btnDoubleTempo</sender>
   <signal>clicked()</signal>
   <receiver>MetronomeWindow</receiver>
   <slot>doubleTempo()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>266</x>
     <y>107</y>
    </hint>
    <hint type="destinationlabel">
     <x>276</x>
     <y>881</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>tempoChanged(int)</signal>
  <slot>startStopMetronome(bool)</slot>
  <slot>changeTempo(int)</slot>
  <slot>chooseSubdivision(QAbstractButton*,bool)</slot>
  <slot>tapTempo()</slot>
  <slot>halfTempo()</slot>
  <slot>doubleTempo()</slot>
 </slots>
 <buttongroups>
  <buttongroup name="bgDivisions"/>
 </buttongroups>
</ui>
